apps/api/src/app/workflows-v2/dto/workflow.dto.ts

```diff
@@ -0,0 +1,109 @@
+import { IsArray, IsBoolean, IsDefined, IsEnum, IsObject, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { Type } from 'class-transformer';
+
+import { JsonSchema } from '@novu/framework';
+import { StepTypeEnum, WorkflowChannelPreferences, WorkflowOriginEnum } from '@novu/shared';
+import { RequiredProp } from '../customTypes';
+
+class ControlsSchema {
+  schema: JsonSchema;
+}
+
+export type CreateWorkflowDto = Omit<WorkflowDto, '_id'>;
+
+export type UpdateWorkflowDto = RequiredProp<Omit<WorkflowDto, '_id'>, 'origin'> & {
+  updatedAt: string;
+  steps: (StepResponseDto | StepDto)[];
+};
+
+export type WorkflowResponseDto = RequiredProp<Omit<WorkflowDto, 'steps'>, 'origin'> & {
+  updatedAt: string;
+  steps: StepResponseDto[];
+
+  /**
+   * Enum to define the origin of the workflow.
+   *
+   * The `WorkflowOriginEnum` is used to specify the source of the workflow,
+   * which helps determine which endpoint to call during the Preview & Execution phase.
+   * - 'novu' indicates that the workflow originates from Novu's platform, so the Novu endpoint is used.
+   * - 'external' indicates that the workflow originates from an external source, requiring a call to a customer-hosted Bridge endpoint.
+   */
+  origin: WorkflowOriginEnum;
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please move this typedoc to the `WorkflowOriginEnum`, so that it can live with the implementation. Folks who need to understand the purpose of `origin` can go there.‏

Reply:
Will do